### PR TITLE
CFIN 281 Add Prettier and XO

### PR DESCRIPTION
**Pull request by ncn1**

### CODE REVIEW

File: .xo-config.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "capitalized-comments": "off",
+    "indent": "off",
+    "import/no-unassigned-import": "off",
+    "react/react-in-jsx-scope": "off",
+    "unicorn/catch-error-name": "off",
+    "unicorn/filename-case": "off",
+    "unicorn/import-index": "off",
+    "unicorn/prevent-abbreviations": "off"
+  },
+  "space": true
```
Comment (FergusMcGlynn) [inline .xo-config.json:24]: I found that I needed to amend this and specify that 4 spaces is our standard (as automated by prettier) or I got XO errors in various places.
```suggestion
  "space": 4
```

**Review by FergusMcGlynn — APPROVED** [review]

File: .xo-config.json (hunk 1 above)
Comment (ncn1) [reply .xo-config.json:24]: Happy to change it, but have not had any of those errors.

**Review by ce666-york — CHANGES_REQUESTED** [review]
I'd like to tie up the uncertainty around using React libraries - all of my suggestions create a version that works locally for me, but I want to check whether there's a specific reason we're including them.

File: package.json [hunk 2]
```diff
@@ -5,15 +5,21 @@
...
+    "lint": "xo ."
   },
   "keywords": [],
   "author": "University of York",
   "license": "MIT",
   "devDependencies": {
-    "serverless": "^2.1.1",
+    "eslint-config-xo-react": "^0.23.0",
+    "eslint-plugin-react": "^7.21.1",
+    "eslint-plugin-react-hooks": "^4.1.2",
```
[the remark this answers is not in the document]
Comment (FergusMcGlynn) [reply package.json:17]: Oh yeah! Good point!

File: .xo-config.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "extends": ["xo-react", "prettier", "prettier/react"],
```
[the remark this answers is not in the document]
Comment (ncn1) [reply .xo-config.json:2]: I copied and pasted from the Course Search,  without registering the API didn't use react. Got a little bogged down with it all, was hoping reviewers would spot stuff like this!

**Comment by ncn1** [comment]
Going to close and create a pull request into CFIN-204 instead.

**Pull request closed without merging**